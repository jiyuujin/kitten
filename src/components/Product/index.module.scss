.basic {
    font-weight: 200;
    font-size: 28px;
    text-transform: uppercase;
    line-height: 1;
    text-shadow: 0 2px 5px rgba(0, 0, 0, 0.3);
    margin: 16px 0 24px 0;
    max-width: 100%;
    text-align: left;

    .subtitle {
        font-size: 1.2rem;
        margin: 18px 0;
    }

    .description {
        font-size: 0.8rem;
        margin: 12px 0;
        line-height: 24px;
    }
}

.link {
    display: flex;
    justify-content: flex-start;

    a {
        margin: 0 8px;
    }
}

.skills {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
    flex: 0 1 auto;
    font-size: 0.8rem;
    line-height: 24px;
}
