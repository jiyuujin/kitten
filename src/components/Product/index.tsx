import React, { FC } from 'react'
import { useIntl } from 'gatsby-plugin-intl'
import * as SC from './index.module.scss'

type ProductProps = {
    data: Array<{
        id: string
        title: string
        description: string
        skills: string[]
    }>
}

const Product: FC<ProductProps> = ({ data }) => {
    const intl = useIntl()

    return (
        <div className={SC.basic}>
            <div className={SC.subtitle}>
                {intl.formatMessage({ id: 'product' })}
            </div>
            {data.map(({ node }: any) => {
                return (
                    <div key={node.id}>
                        <div className={SC.subtitle}>{node.title}</div>
                        <div
                            className={SC.description}
                            dangerouslySetInnerHTML={{
                                __html: intl.formatMessage({
                                    id: node.description,
                                }),
                            }}
                        />
                        <div className={SC.subtitle}>
                            {intl.formatMessage({
                                id: 'technology_used',
                            })}
                        </div>
                        <div className={SC.description}>
                            <ul>
                                {node.skills.map((skill: string) => {
                                    return <li key={skill}>{skill}</li>
                                })}
                            </ul>
                        </div>
                    </div>
                )
            })}
        </div>
    )
}

export default Product
