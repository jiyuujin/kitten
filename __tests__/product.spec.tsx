import React from 'react'
import renderer from 'react-test-renderer'
import { IntlContextProvider } from 'gatsby-plugin-react-intl'
import messages from '../src/intl/en.json'
import Product from '../src/components/Product'

const imageData = [
    {
        node: {
            id: 'abc',
            title: 'title',
            description: 'title',
            skills: ['title'],
            image: 'title',
        },
    },
]

const locales = ['en', 'ja']

jest.mock('react-intl', () => {
    const reactIntl = jest.requireActual('react-intl')
    const intl = reactIntl.createIntl({
        locale: 'en',
    })
    return {
        ...reactIntl,
        useIntl: () => intl,
    }
})

describe('Product', () => {
    it('render correctly', () => {
        const tree = renderer
            .create(
                <IntlContextProvider
                    value={{
                        language: 'en',
                        languages: locales,
                        messages,
                        originalPath: '/',
                        redirect: true,
                        routed: true,
                    }}
                >
                    <Product data={imageData} />
                </IntlContextProvider>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
