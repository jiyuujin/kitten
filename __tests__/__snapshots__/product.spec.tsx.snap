// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product render correctly 1`] = `
<div>
  <div>
    product
  </div>
  <div>
    <div>
      title
    </div>
    <img
      alt="title"
      decoding="async"
      src="title"
    />
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "title",
        }
      }
    />
    <div>
      technology_used
    </div>
    <div>
      <ul>
        <li>
          title
        </li>
      </ul>
    </div>
  </div>
</div>
`;
